graphein/protein/visualisation.py

```diff
@@ -34,13 +37,215 @@ def plot_pointcloud(mesh: Meshes, title: str = "") -> None:
+    colour_nodes_by: str = "degree",
+    colour_edges_by: str = "type",
+    edge_alpha: float = 0.5,
+    out_path: Optional[str] = None,
+    out_format: str = ".png",
+):
+    """
+    Plots protein structure graph in Axes3D.
+    :param G:  nx.Graph Protein Structure graph to plot
+    :param angle:  View angle
+    :param figsize: Size of figure
+    :param node_alpha: Controls node transparency
+    :param node_size_min: Specifies node minimum size
+    :param node_size_multiplier: Scales node size by a constant. Node sizes reflect degree.
+    :param label_node_ids: bool indicating whether or not to plot node_id labels
+    :param node_colour_map: colour map to use for nodes
+    :param edge_color_map: colour map to use for edges
+    :param colour_nodes_by: Specifies how to colour nodes. "degree"m "seq_position" or a node eature
+    :param colour_edges_by: Specifies how to colour edges. Currently only "kind" is supported
+    :param edge_alpha: Controls edge transparency
+    :param out_path: If not none, writes plot to this location
+    :param out_format: Fileformat to use for plot
+    :return:
+    """
+    # Get Node Attributes
+    pos = nx.get_node_attributes(G, "coords")
+
+    # Get node colours
+    node_colors = colour_nodes(
+        G, colour_map=node_colour_map, colour_by=colour_nodes_by
+    )
+    edge_colors = colour_edges(
+        G, colour_map=edge_color_map, colour_by=colour_edges_by
+    )
+
+    # 3D network plot
+    with plt.style.context(("ggplot")):
+
+        fig = plt.figure(figsize=figsize)
+        ax = Axes3D(fig)
+
+        # Loop on the pos dictionary to extract the x,y,z coordinates of each node
+        for i, (key, value) in enumerate(pos.items()):
+            xi = value[0]
+            yi = value[1]
+            zi = value[2]
+
+            # Scatter plot
+            ax.scatter(
+                xi,
+                yi,
+                zi,
+                color=node_colors[i],
+                s=node_size_min + node_size_multiplier * g.degree[key],
+                edgecolors="k",
+                alpha=node_alpha,
+            )
+            if label_node_ids:
+                label = list(G.nodes())[i]
+                ax.text(xi, yi, zi, label)
+
+        # Loop on the list of edges to get the x,y,z, coordinates of the connected nodes
+        # Those two points are the extrema of the line to be plotted
+        for i, j in enumerate(g.edges()):
+            x = np.array((pos[j[0]][0], pos[j[1]][0]))
+            y = np.array((pos[j[0]][1], pos[j[1]][1]))
+            z = np.array((pos[j[0]][2], pos[j[1]][2]))
+
+            # Plot the connecting lines
+            ax.plot(x, y, z, c=edge_colors[i], alpha=edge_alpha)
+
+    # Set the initial view
+    ax.view_init(30, angle)
+    # Hide the axes
+    ax.set_axis_off()
+    if out_path is not None:
+        plt.savefig(out_path + str(angle).zfill(3) + out_format)
+        plt.close("all")
+    else:
+        plt.show()
+
+    return
```

Comment:
I'd suggest returning the matplotlib axes object, rather than calling on plt.show(), so that end-users can do further manipulations to it. (For example, rotating the view, adding labels, etc.)

Reply:
Absolutely, great idea